### PR TITLE
Addition of pixelPairStep

### PR DESCRIPTION
**Pull request by mmasciov**
For internal consumption.
For PR to CMSSW, we could use directly: https://github.com/trackreco/cmssw/tree/mkFitPixelPairStep
together with https://github.com/mmasciov/RecoTracker-MkFit/tree/mkFitPixelPairStep

This PR is related to trackreco/mkFit#320

Note: in this PR, I'm also commenting out the sanity check on seed hits, to be able to run with latest mkFit (including pixel seed merging)

### CODE REVIEW

File: RecoTracker/IterativeTracking/python/PixelPairStep_cff.py [hunk 1]
```diff
@@ -298,6 +298,30 @@
     onlyPixelHitsForSeedCleaner = cms.bool(True),
 
 )
+
+from Configuration.ProcessModifiers.trackingMkFitPixelPairStep_cff import trackingMkFitPixelPairStep
+import RecoTracker.MkFit.mkFitSeedConverter_cfi as mkFitSeedConverter_cfi
+import RecoTracker.MkFit.mkFitIterationConfigESProducer_cfi as mkFitIterationConfigESProducer_cfi
+import RecoTracker.MkFit.mkFitProducer_cfi as mkFitProducer_cfi
+import RecoTracker.MkFit.mkFitOutputConverter_cfi as mkFitOutputConverter_cfi
```
Comment (makortel) [inline RecoTracker/IterativeTracking/python/PixelPairStep_cff.py:306]: I'd suggest
```suggestion
import RecoTracker.MkFit.mkFitSeedConverter_cfi as _mkFitSeedConverter_cfi
import RecoTracker.MkFit.mkFitIterationConfigESProducer_cfi as _mkFitIterationConfigESProducer_cfi
import RecoTracker.MkFit.mkFitProducer_cfi as _mkFitProducer_cfi
import RecoTracker.MkFit.mkFitOutputConverter_cfi as _mkFitOutputConverter_cfi
```
in light of Joosep's comment in the `initialStepPreSplitting` PR.

**Comment by makortel** [comment]
> I'm also commenting out the sanity check on seed hits, to be able to run with latest mkFit (including pixel seed merging)

That basically means that the downstream will see a random seed associated for each track (and an exception if the mkFit's `cand.label()` exceeds the number of seeds in CMSSW's seed container). It could be acceptable for near term to allow development to proceed. @slava77, what do you think?

File: RecoTracker/IterativeTracking/python/PixelPairStep_cff.py (hunk 1 above)
Comment (mmasciov) [reply RecoTracker/IterativeTracking/python/PixelPairStep_cff.py:306]: Will do and push. Thanks

**Comment by mmasciov** [comment]
@makortel and @slava77: I am OK with keeping the sanity check, but if we keep it and make a distribution with the current mkFit to be used within CMSSW, then it will crash after N events (when pixel seed merging takes place). If we wish to keep the check, we should probably solve the source of the crash (in the seed merging procedure), or the code will not really be usable.

**Comment by makortel** [comment]
In my opinion the seed issue should be addressed properly (whatever that is) before making a new release for CMS.

**Comment by mmasciov** [comment]
I have reverted the change in MkFitOutputConverter (i.e., restored the sanity check) in branch trackreco/cmssw/mkFitPixelPairStep, for eventual PR to CMSSW.
Can restore it here too, while the issue is properly addressed.

**Comment by slava77** [comment]
> That basically means that the downstream will see a random seed associated for each track (and an exception if the mkFit's `cand.label()` exceeds the number of seeds in CMSSW's seed container). It could be acceptable for near term to allow development to proceed. @slava77, what do you think?

as I mentioned earlier on skype, I did not quite follow the implications.
I thought that after merging several seeds we simply get a "randomly" selected label of one of the contributing original seeds. If that's the case, then it should be just fine to use this label to set the output TrackCandidate seed and seedRef using the original seed with such label. However, if the point is that the output TrackCandidate will have a nonsense seed/seedRef, then it's bad.

**Comment by makortel** [comment]
To what does the cand.label() point to when seeds are merged? Is it something in the original seed container, or something in the container of merged seeds (if such exists)? In case if the former, disabling the sanity check would be ok (although maybe make it configurable?), I have been assuming the latter.

**Comment by mmasciov** [comment]
We used to create a container of clean seeds, after seed cleaning, which is a subset of the original seed collection (from CMSSW).
The only difference after trackreco/mkFit#321 is that [we now add hits from overlapping seeds to seeds in the same collection] (https://github.com/trackreco/mkFit/pull/321/files#diff-bbf7f7daabbb2c72e01c99bc493c8384d96badb4f375a3b1e71502ddf4bd8f83R282). The label/index of the seed should not be affected (@osschar).

**Comment by makortel** [comment]
On a further thought, if the `cand.label()` is guaranteed to point to an index in one of the merged seeds in the original seed container, I think the "sanity check" could actually be removed as it is not that useful anymore (even without merging).

**Comment by mmasciov** [comment]
Do you prefer to leave it as is (commented out) for history, or removed completely for cleaning purposes?
If the latter, I'll just remove it.

**Comment by makortel** [comment]
I'd prefer to remove it completely (history stays in git anyway). If there would be any motivation to keep the implementation, I'd make it configurable.

**Comment by mmasciov** [comment]
As per latest commit, the sanity check has been removed (I don't see a reason to keep it as configurable at this stage).

**Comment by makortel** [comment]
Looks good now, thanks. Could you then squash the commits into two, one for removing the sanity check, and second for adding the pixelPairStep? That would keep the history cleaner and help for any future rebase of the `CMSSW_11_2_0_mkFit_X` and PR(s) towards CMSSW (although you already have a branch for the latter).

**Comment by mmasciov** [comment]
Commits have been squashed as suggested (@makortel).

**Comment by makortel** [comment]
Thanks!

**Pull request merged**